Don't suppress HDR in pseudo full screen videos

squelart opened the pull request:
#### e954f76dfe8c0802fc102451ba251b7422d5c109
<pre>
Don&apos;t suppress HDR in pseudo full screen videos
<a href="https://bugs.webkit.org/show_bug.cgi?id=291474">https://bugs.webkit.org/show_bug.cgi?id=291474</a>
<a href="https://rdar.apple.com/149128748">rdar://149128748</a>

Reviewed by Jean-Yves Avenard.

Some sites like YouTube don&apos;t make the video element full screen, but
a full screen document with an enlarged video element.
VideoPresentationModelVideoElement can already detect this, and forwards
full screen changes to HTMLMediaElement, to decide whether to suppress
HDR; when the video is in full screen (directly or indirectly), never
suppress HDR.

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::documentFullscreenChanged):
(WebCore::HTMLMediaElement::computePlayerDynamicRangeLimit const):
* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm:
(WebCore::VideoPresentationModelVideoElement::documentFullscreenChanged):

Canonical link: <a href="https://commits.webkit.org/293653@main">https://commits.webkit.org/293653@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0489edd44a3150049f03e9f490bf068242e33728

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99522 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/19172 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/9426 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104653 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/50124 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/101563 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/19460 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27605 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75755 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32857 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102529 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14818 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89863 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/56114 "Found 1 new API test failure: /WPE/TestSSL:/webkit/WebKitWebView/ephemeral-tls-errors (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14615 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7848 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49483 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84548 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7934 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/107011 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26636 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/19442 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84715 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26998 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/86067 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84232 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21372 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28905 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6603 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/20427 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26576 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/31777 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/26396 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29709 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27963 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on Source/WebCore/html/HTMLMediaElement.h:
```diff
@@ -1225,6 +1229,9 @@ class HTMLMediaElement
     VideoFullscreenMode m_videoFullscreenMode { VideoFullscreenModeNone };
     bool m_videoFullscreenStandby { false };
+#if ENABLE(FULLSCREEN_API)
+    bool m_isChildOfDocumentFullscreenElement { false };
```
```suggestion
    bool m_isChildOfElementFullscreen { false };
```

to be in line with the nomenclature used in the file and typically when referred to it..

jyavenard reviewed (approved):
r=me with name change

squelart committed:
Don't suppress HDR in pseudo full screen videos
https://bugs.webkit.org/show_bug.cgi?id=291474
rdar://149128748

Reviewed by Jean-Yves Avenard.

Some sites like YouTube don't make the video element full screen, but
a full screen document with an enlarged video element.
VideoPresentationModelVideoElement can already detect this, and forwards
full screen changes to HTMLMediaElement, to decide whether to suppress
HDR; when the video is in full screen (directly or indirectly), never
suppress HDR.

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::documentFullscreenChanged):
(WebCore::HTMLMediaElement::computePlayerDynamicRangeLimit const):
* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm:
(WebCore::VideoPresentationModelVideoElement::documentFullscreenChanged):

Canonical link: https://commits.webkit.org/293653@main
